AEPCore/Sources/core/MobileCoreInitializer.swift

```diff
@@ -0,0 +1,111 @@
+/*
+ Copyright 2025 Adobe. All rights reserved.
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+import AEPServices
+import Foundation
+import UIKit
+
+final class MobileCoreInitializer {
+    private let LOG_TAG = "MobileCoreInitializer"
+
+    // Flag if 'initialize()' was called
+    private let initialized = AtomicCounter()
+
+#if DEBUG
+    // Function to return list of classes; (Protocol) -> [AnyClass]
+    public internal(set) var classFinder = ClassFinder.classes(conformToProtocol:)
+#else
+    // Function to return list of classes; (Protocol) -> [AnyClass]
+    private let classFinder = ClassFinder.classes(conformToProtocol:)
+#endif
+
+#if DEBUG
+    public internal(set) static var shared = MobileCoreInitializer()
+#else
+    static let shared = MobileCoreInitializer()
+#endif
+
+    @available(iOSApplicationExtension, unavailable)
+    @available(tvOSApplicationExtension, unavailable)
+    func initialize(options: InitOptions, _ completion: (() -> Void)? = nil) {
+
+        if initialized.incrementAndGet() != 1 {
+            Log.debug(label: LOG_TAG, "initialize - ignoring as it was already called.")
+            return
+        }
+
+        if options.appGroup != nil {
+            MobileCore.setAppGroup(options.appGroup)
+        }
+
+        if let appId = options.appId {
+            MobileCore.configureWith(appId: appId)
+        } else if let filePath = options.filePath {
+            MobileCore.configureWith(filePath: filePath)
+        }
+
+        // Register Extensions
+        DispatchQueue.global().async {
+            let classList = self.classFinder(Extension.self)
+            let filteredClassList = classList.filter { $0 !== AEPCore.EventHubPlaceholderExtension.self && $0 !== AEPCore.Configuration.self }.compactMap { $0 as? NSObject.Type }
+            MobileCore.registerExtensions(filteredClassList) {
+                // If lifecycleAutomaticTracking flag is false, set lifecycle notification listeners
```

Comment:
I also prefer moving the automatic lifecycle tracking before calling registerExtensions so that we don't delay the call. We initially had it this way because in ACP events were dropped before the event hub was initialized, which is not the case with AEP.